Sources/TSFFutures/BatchingFutureOperationQueue.swift

```diff
@@ -51,49 +84,72 @@ public struct LLBBatchingFutureOperationQueue {
-        operationQueue.addOperation {
-            promise.fulfill(body)
+        return self.concurrencyLimiter.withReplenishableLimit(eventLoop: group.any()) { eventLoop in
+            let promise = eventLoop.makePromise(of: T.self)
+
+            DispatchQueue(label: self.name, qos: self.qos).async {
+                promise.fulfill(body)
+            }
+
+            return promise.futureResult
         }
-        return promise.futureResult
     }
 
-    @inlinable
     public func execute<T>(_ body: @escaping () -> LLBFuture<T>) -> LLBFuture<T> {
-        let promise = group.next().makePromise(of: T.self)
-        operationQueue.addOperation {
-            let f = body()
-            f.cascade(to: promise)
+        return self.concurrencyLimiter.withReplenishableLimit(eventLoop: group.any()) { eventLoop in
+            let promise = eventLoop.makePromise(of: T.self)
+
+            DispatchQueue(label: self.name, qos: self.qos).async {
+                body().cascade(to: promise)
+            }
 
-            // Wait for completion, to ensure we maintain at most N concurrent operations.
-            _ = try? f.wait()
+            return promise.futureResult
         }
-        return promise.futureResult
     }
 
     /// Order-preserving parallel execution. Wait for everything to complete.
     @inlinable
-    public func execute<A,T>(_ args: [A], minStride: Int = 1, _ body: @escaping (ArraySlice<A>) throws -> [T]) -> LLBFuture<[T]> {
+    public func execute<A, T>(_ args: [A], minStride: Int = 1, _ body: @escaping (ArraySlice<A>) throws -> [T])
+        -> LLBFuture<[T]>
+    {
         let futures: [LLBFuture<[T]>] = executeNoWait(args, minStride: minStride, body)
         let loop = futures.first?.eventLoop ?? group.next()
-        return LLBFuture<[T]>.whenAllSucceed(futures, on: loop).map{$0.flatMap{$0}}
+        return LLBFuture<[T]>.whenAllSucceed(futures, on: loop).map { $0.flatMap { $0 } }
     }
 
     /// Order-preserving parallel execution.
     /// Do not wait for all executions to complete, returning individual futures.
     @inlinable
-    public func executeNoWait<A,T>(_ args: [A], minStride: Int = 1, maxStride: Int = Int.max, _ body: @escaping (ArraySlice<A>) throws -> [T]) -> [LLBFuture<[T]>] {
-        let batches: [ArraySlice<A>] = args.tsc_sliceBy(maxStride: max(minStride, min(maxStride, args.count / maxOpCount)))
-        return batches.map{arg in execute{try body(arg)}}
+    public func executeNoWait<A, T>(
+        _ args: [A], minStride: Int = 1, maxStride: Int = Int.max, _ body: @escaping (ArraySlice<A>) throws -> [T]
+    ) -> [LLBFuture<[T]>] {
+        let batches: [ArraySlice<A>] = args.tsc_sliceBy(
+            maxStride: max(minStride, min(maxStride, args.count / maxOpCount)))
+        return batches.map { arg in execute { try body(arg) } }
     }
 
+    private static func bridged(maxOpCount: Int) -> Int {
+        switch maxOpCount {
+        case OperationQueue.defaultMaxConcurrentOperationCount:
```

Comment:
Shouldn't it be `<` comparison rather than `==`?

Reply:
@OmSaran what if macOpCount is 2983742, what will this code do?

Reply:
Document this.